### DISCUSSION TITLE
Disabling user login on specific client

### DISCUSSION THREAD

**Discussion by jmurrayrs** [keycloak/keycloak#29940] [Q&A] [answered]
Hello! I am currently working with version 24.0.1 of Keycloak and I want to make the following configuration.
Let's say I have two clients within a realm and in this realm I have several users.
In one of these clients, I want to disable the option for users to log in. How do I do that?

Thanks !

**Comment by danielFesenmeyer** [comment]
Just unselect all authentication flows when creating the client:

![create-client-without-authentication-flows](https://github.com/keycloak/keycloak/assets/11198630/c020de1e-0f03-4869-ac2d-181c26ad1cd0)

**Reply by jmurrayrs** [reply]
Hello @danielFesenmeyer ! Thank you for your help. I'll be more specific.

Okay, your example solve the problem, but If I want to use this client to login a web-api or a worker, the authentication will fail.

My problems is:

I have a Realm

In this Realm I have :

worker-1 and web-api-2 where

worker-1 is a service, and this service is used to do something, and the worker-1 uses keycloak to only authenticate and get the token.

web-api-2 will receive user login where the user will do the flow authentication.

If I use this URL `/realms/myrealm/protocol/openid-connect/auth?response_type=code&client_id=worker-1&redirect_uri=xpto=random_state&scope=openid`, my user I'll get access using the worker client being that the correct client is web-api-2.

I have read the documentation, but I can't find the solution for it.

Thanks Bro !

**Reply by jmurrayrs** [reply, marked as answer]
I'm not sure if this is the best approach, but I found a good solution to my problem [here](https://stackoverflow.com/questions/54305880/how-can-i-restrict-client-access-to-only-one-group-of-users-in-keycloak).

**_"On the Keycloak admin console, go to the Clients menu, select your client. On the client configuration page, set Authorization Enabled to On and click Save. A new Authorization tab should appear; go to it, then to the Policies tab, click Create Policy, and select Group-based policy. There, you can restrict access to specific groups, assuming you have defined your groups via the Groups menu already.
As mentioned in the comments, Client Protocol must be set to openid-connect and Access Type must be set to confidential in order to make the Authorization Enabled option visible."_**

Here's what I did:

In Realm Settings > Client Policies > Policies, I created a policy called BlockLoginAccess. Under Conditions, I created two items:

- client-access-type, where Client Access Type is set to bearer-only
- client-roles, where Client Roles is set to DenyLoginAccess

I saved the changes.

In Clients, I selected the client where I don't want users to log in. Then I went to Authorization > Policies > Create Client Policy.

In the policy name, I entered DenyLoginAccess and added the user groups I want to deny access to. In this case, I created a root group called users and set Logic to "Negative."

Now, users can no longer log in to my client.

**Reply by lgwapnitsky** [reply]
Not working as of the latest version of Keycloak

**Comment by vi-buuuk** [comment]
Hi @jmurrayrs 

I'm using keycloak 25.0.4, I didn't see client policy inside 

Authorization, Can you help me check it ?

<img width="835" alt="Screenshot 2024-09-12 at 15 56 02" src="https://github.com/user-attachments/assets/64334741-a948-40e2-9657-86a5ca7e028e">
